### PR TITLE
ADD the URL Embed Text Filter

### PR DESCRIPTION
**Pull request by cs-shadow**
Working on #1 to make the tests work.

### CODE REVIEW

**Comment by cs-shadow** [comment]
The test fails because Embed is returning error status 429. Tests should pass locally.

@prateekmehta  @davereid  Would appreciate if you both can test it locally before we merge this since the tests are failing on Travis atm.

File: src/Plugin/Filter/UrlEmbedFilter.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+  /**
+   * Replace the contents of a DOMNode.
+   *
+   * @param \DOMNode $node
+   *   A DOMNode or DOMElement object.
+   * @param string $content
+   *   The text or HTML that will replace the contents of $node.
+   */
+  protected function setDomNodeContent(\DOMNode $node, $content) {
```
Comment (davereid) [inline src/Plugin/Filter/UrlEmbedFilter.php:37]: Noting that as soon as possible we should add an Embed module dependency and re-use the function from DomHelperTrait

**Comment by davereid** [comment]
Looks like we currently have a merge conflict that needs to be resolved. Ideally I'd like to get this committed once we also have the internal module we can use for testing embedding.

File: src/Plugin/Filter/UrlEmbedFilter.php (hunk 1 above)
Comment (cs-shadow) [inline src/Plugin/Filter/UrlEmbedFilter.php:37]: Created a followup just so that this is not overlooked: https://www.drupal.org/node/2547105

**Comment by cs-shadow** [comment]
@davereid Fixed the conflicts and ran the tests again locally to ensure it still works. Although ideally we should wait for the test module but I suggest merging it now so that other issues (like ckeditor integration) are not blocked by it.

**Comment by davereid** [comment]
Yeah, I just don't want to get too far into the CKEditor integration without having the other parts ready to go, like field formatters (and hence display plugins).

File: scripts/composer-update.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e $DRUPAL_TI_DEBUG
+
+# Ensure the right Drupal version is installed.
+# Note: This function is re-entrant.
+drupal_ti_ensure_drupal
+
+cd "$DRUPAL_TI_DRUPAL_DIR"
+cd core
```
Comment (davereid) [inline scripts/composer-update.sh]: This could be just written as `cd "$DRUPAL_TI_DRUPAL_DIR\core"` ?

File: scripts/composer-update.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+composer drupal-rebuild
+composer update embed --prefer-source
```
Comment (davereid) [inline scripts/composer-update.sh:11]: Why is this necessary (just curious)?

File: src/Plugin/Filter/UrlEmbedFilter.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+use Drupal\filter\Plugin\FilterBase;
+use Embed\Embed;
+
+/**
+ * Provides a filter to display embedded URLs based on data attributes.
+ *
+ * @Filter(
+ *   id = "url_embed",
+ *   title = @Translation("Display embedded URLs"),
+ *   description = @Translation("Embeds URLs using data attributes: url, source, oembed provider and view-mode."),
```
Comment (davereid) [inline src/Plugin/Filter/UrlEmbedFilter.php]: References view-mode here, remove.

File: scripts/composer-update.sh (hunk 3 above)
Comment (cs-shadow) [inline scripts/composer-update.sh:11]: Sometimes travis fails to download packages from Github, due to API limit or whatever. This option helps avoid that. One issue which I referenced: https://github.com/composer/composer/issues/1314

**Comment by cs-shadow** [comment]
@davereid Addressed your comments and also replaced youtube url for testing with a fickr url which seems to work fine. And from what it seems flickr doesn't blacklist travis often so we should be good.

The tests are now passing on Travis but fail on PHP 7 for no apparent reason. It looks like something wrong with our custom script but I suggest to get this PR in so that @prateekmehta is not blocked by it.

File: src/Plugin/Filter/UrlEmbedFilter.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+use Drupal\filter\Plugin\FilterBase;
+use Embed\Embed;
+
+/**
+ * Provides a filter to display embedded URLs based on data attributes.
+ *
+ * @Filter(
+ *   id = "url_embed",
+ *   title = @Translation("Display embedded URLs"),
+ *   description = @Translation("Embeds URLs using data attribute: data-url."),
```
Comment (davereid) [inline src/Plugin/Filter/UrlEmbedFilter.php]: Should have noted earlier, the attribute is actually data-embed-url

**Comment by cs-shadow** [comment]
@davereid Any idea whats up with PHP 7 build?

**Comment by Berdir** [comment]
https://www.drupal.org/node/2547715 is up with PHP 7.

**Comment by cs-shadow** [comment]
@Berdir thanks! That makes more sense, I thought I broke the build again right after I barely got it to work.
